### PR TITLE
Adding Network Profile options for AKS clusters

### PR DESCRIPTION
**Pull request by evillgenius75**
In response to issue #238 - Added the proper fields for the AKS clusters to deploy with additional Network Profile options such as serviceCIDR, DNS_Service_IP, Docker_Bridge_CIDR and to also include custom IP address for VNet and Subnet addresses. These network profile options will be required when joining an AKS environment to a hybrid connected VNet such as VPN, VNet Peered or ExpressRoute connected set-up.

Modified original AKS module and added the needed entries in each of the environment samples to properly deploy. Tested all 3 environment setups. Also increased AzureRm provider to latest 1.23.1

Eddie Villalba
Principal SE - Microsoft 
edvilla@microsoft.com

### CODE REVIEW

File: cluster/environments/azure-simple/terraform.tfvars [hunk 1]
```diff
@@ -5,6 +5,11 @@ resource_group_location="westus2"
 cluster_name="cluster-name"
 agent_vm_count = "3"
 dns_prefix="dns-prefix"
+address_space="172.20.0.0/16"
+subnet_prefixes=["172.20.0.0/20"]
+service_CIDR="172.21.0.0/16"
+dns_IP="172.21.0.10"
```
Comment (dtzar) [inline cluster/environments/azure-simple/terraform.tfvars]: These values do not match up to the default vars below.  In general, would prefer to keep the `azure-simple` have as little possible needed configurable values as possible, so perhaps just set the defaults to the same as the defaults from the aks module in the `variables.tf` file and remove what's here in the `terraform.tfvars` file.

File: cluster/azure/aks/main.tf [hunk 2]
```diff
@@ -33,6 +33,9 @@ resource "azurerm_kubernetes_cluster" "cluster" {
 
   network_profile {
     network_plugin = "azure"
+    service_cidr = "${var.service_CIDR}"
```
[the remark this answers is not in the document]
Comment (evillgenius75) [reply cluster/azure/aks/main.tf]: Sounds good on both counts, I will make the changes and recommit

**Comment by dtzar** [comment]
Build failing
```
span>TestIT_Bedrock_AzureSimpcommand.go:121: ‌
TestIT_Bedrock_AzureSimpcommand.go:121: ‌Error: ‌Error loading &#x2F;home&#x2F;vsts&#x2F;work&#x2F;1&#x2F;s&#x2F;gopath&#x2F;src&#x2F;github.com&#x2F;Microsoft&#x2F;bedrock&#x2F;cluster&#x2F;environments&#x2F;azure-simple&#x2F;main.tf: Error reading config for vnet: At 18:29: unknown slice type: *ast.LiteralType‌
```

**Comment by timfpark** [comment]
@NathanielRose Do you understand what the issue is here with the build break that Azure Devops is reporting?

**Pull request merged**